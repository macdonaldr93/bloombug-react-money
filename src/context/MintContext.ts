import { createContext } from 'react';
import { Mint } from '@bloombug/money';

export interface IMintContext {
  mint: Mint;
}

export const MintContext = createContext<IMintContext>({ mint: new Mint() });
