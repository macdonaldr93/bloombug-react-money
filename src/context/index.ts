export { IMintContext } from './MintContext';
export { MintProvider } from './MintProvider';
