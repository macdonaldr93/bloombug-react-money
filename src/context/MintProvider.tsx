import React, { FC, ReactChild } from 'react';
import { Mint } from '@bloombug/money';

import { MintContext } from './MintContext';

export interface Props {
  children: ReactChild;
  mint: Mint;
}

export const MintProvider: FC<Props> = ({ children, mint }) => {
  return (
    <MintContext.Provider value={{ mint }}>{children}</MintContext.Provider>
  );
};
