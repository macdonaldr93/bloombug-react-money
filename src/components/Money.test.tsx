import React from 'react';
import * as ReactDOM from 'react-dom';
import { CAD, Mint } from '@bloombug/money';
import currencies from '@bloombug/money/iso-currencies.json';

import { MintProvider } from '../context';
import { Money } from './Money';

describe('<Money />', () => {
  let div: HTMLDivElement;

  beforeEach(() => {
    div = document.createElement('div');
  });

  afterEach(() => {
    ReactDOM.unmountComponentAtNode(div);
  });

  it('returns default format money', () => {
    ReactDOM.render(<Money fractional={100} />, div);

    expect(div).toMatchSnapshot();
  });

  it('returns formatted money', () => {
    const mint = new Mint({ currencies });

    ReactDOM.render(
      <MintProvider mint={mint}>
        <Money fractional={100} currency={CAD} locale="fr-FR" />
      </MintProvider>,
      div
    );

    expect(div).toMatchSnapshot();
  });
});
