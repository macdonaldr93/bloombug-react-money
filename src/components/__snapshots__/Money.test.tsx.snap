// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Money /> returns default format money 1`] = `
<div>
  <span>
    US$1.00
  </span>
</div>
`;

exports[`<Money /> returns formatted money 1`] = `
<div>
  <span>
    1,00Â $CA
  </span>
</div>
`;
